### PR TITLE
Add ccls configuration to package settings

### PR DESCRIPTION
**Pull request by xphoenix**
Add most of ccls init options into the package config file, so that it now could be set per project file (via project-manager):
```
settings:
    "*":
        "ide-ccls":
            cclsInit:
              compilationDatabaseCommand:"./generate_compile_database.sh",
              index:
                blacklist: [".*"],
                whitelist: ["src/*","include/*","mock/*","utest/*"]
```

### CODE REVIEW

**Comment by xphoenix** [comment]
I've failed to add 2 things:
- Collapse options by default (really, has no idea how to make it :O)
- Need some love on initialization in order to get ccls version properly (mine reports 7.0.1 which IMHO clang version) and render config. For example cacheDirectly has changed its path in the last ccls version

**Comment by isundaylee** [comment]
Hi @xphoenix, thanks for contributing! This looks great! 

For your first point, I have no problem with having the options expanded by default, as long as we keep them at the end so that they don't overwhelm the other options that come before. 

The second point is trickier. Do you happen to know what `ccls` does when it encounters an option that it doesn't recognize (say, passing a new option to an old version)?

**Comment by xphoenix** [comment]
>do you happen to know what `ccls` does when it encounters an option that it doesn't recognize (say, passing 
I've made quick test, ccls just ignores unknown property. Im now trying to prove it throught it's codebase...

**Comment by isundaylee** [comment]
In that case it's probably fine the way it is. In the future, we can probably do something like this? 

- Get `ccls` to print some warnings upon unknown options.
- We parse the warnings and get the list of unknown options. For each option, if the current value is different from the default value (i.e. the user has changed the option value), we issue a warning dialog box.

Let me know when you're happy with the state of the PR, and I'll merge it. By the way, I just added a change log file in `master`. It'd be great if you can rebase and write a short entry in the `master` section of the change log. Thanks :)

**Comment by isundaylee** [comment]
One minor thing: We might wanna remove the option `highlight.lsRanges` since currently we only support using position in semantic highlighting messages.

File: lib/config.js [hunk 1]
```diff
@@ -6,12 +6,467 @@ module.exports = {
     default: "ccls",
     order: 0,
   },
-  indexOnChange: {
-    type: "boolean",
-    title: "Index on file change",
-    description: "Tell ccls to index immediately when a file changes.",
-    default: false,
-    order: 10,
+  cclsInit: {
```
Comment (isundaylee) [inline lib/config.js:9]: Maybe move to the end and add `order: 1000`? That way we can keep it expanded by default.

File: lib/config.js [hunk 2]
```diff
@@ -6,12 +6,467 @@ module.exports = {
...
+            title: "Detailed label",
+            order: 3,
+            description: "When this option is enabled, the completion item label shows the full signature of the candidate.",
+            default: true
+          },
+          dropOldRequests: {
+            type: "boolean",
+            title: "Drop old requests",
+            order: 4,
+            description: "Preffer to service only newest requests.",
```
Comment (isundaylee) [inline lib/config.js]: Prefer? Might be a typo from `ccls` itself.

**Comment by xphoenix** [comment]
> We might wanna remove the option `highlight.lsRanges`
done

> add `order: 1000`
done

> Might be a typo from `ccls` itself.
Nope, its mine. done :)

> In the future, we can probably do something like this
I just start reading through ccls codebase. First of all I want to find a way how to get proper version to know what config options supported. But your suggestion looks reasonable too, will add to the list.

So far Im happy how it is looks like, removed all "local patches" from ide-ccls and configured all my projects individually.

**Comment by isundaylee** [comment]
Merged! I'm currently waiting for some auto-completion bug fixes in `atom-languageclient` to be accepted. After I pull in those fixes I'll release a new version containing this PR as well as the bug fixes. I can also publish a new version with just this if you want it sooner - just let me know. Thanks for doing this - it looks great!

**Comment by isundaylee** [comment]
By the way I just noticed that after "squash and merge" GitHub changed the author name from `Philipp Andronov` to `xphoenix`. Let me know if you want me to change it back.

**Comment by xphoenix** [comment]
Thanks!
Well, for ME it works already, so Im wouldn't ask to release it ASAP. :)
Name is not a big deal

PS:  Greate extension btw, finally I have C++ ide that could be used

**Comment by isundaylee** [comment]
@xphoenix I'm planning on releasing a new version. Do you mind if I add your GitHub username (`xphoenix`) to the change log as the contributor of this PR?

**Pull request merged**